lib/array_intersection.rb

```diff
@@ -1,3 +1,27 @@
+# Time complexity: O(n^2)
+# Space complexity: O(n)
+
 def intersection(list1, list2)
```

Comment:
Your time complexity is O(n + m) where n and m are the lengths of the two arrays.  Otherwise well done.